lib/extensions/ar_to_model_hash.rb

```diff
@@ -110,7 +110,14 @@ def to_model_hash_build_preload(options)
     columns  = ((options && options[:columns]) || [])
     includes = ((options && options[:include]) || {})
 
-    result  = columns.select { |c| self.class.virtual_column?(c) }
+    result = columns.select do |c|
+      next unless self.class.virtual_column?(c)
+
+      # preload the virtual field's uses if they're not already included
+      uses = Array.wrap(self.class.virtual_field(c).uses)
+      (uses - includes.keys).empty?
```

Comment:
Why is this needed?  Just curious because ActiveRecord (at least used to) handle duplicate includes just fine.